src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/FileSystemEgressExtensionTests.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using Microsoft.Diagnostics.Tools.Monitor;
+using Microsoft.Diagnostics.Tools.Monitor.Egress;
+using Microsoft.Diagnostics.Tools.Monitor.Egress.Configuration;
+using Microsoft.Diagnostics.Tools.Monitor.Egress.FileSystem;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.UnitTests
+{
+    public sealed class FileSystemEgressExtensionTests
+    {
+        const string ProviderName = "TestProvider";
+        const string ExpectedFileName = "EgressedData.txt";
+        const string ExpectedFileContent = "This is egressed data.";
+
+        private static readonly string CopyBufferSize_RangeErrorMessage = CreateRangeMessage<int>(
+            nameof(FileSystemEgressProviderOptions.CopyBufferSize),
+            FileSystemEgressProviderOptions.CopyBufferSize_MinValue.ToString(),
+            FileSystemEgressProviderOptions.CopyBufferSize_MaxValue.ToString());
+
+        private readonly ITestOutputHelper _outputHelper;
+
+        public FileSystemEgressExtensionTests(ITestOutputHelper outputHelper)
+        {
+            _outputHelper = outputHelper;
+        }
+
+        [Fact]
+        public async Task FileSystemEgressExtension_EmptyConfiguration_ThrowsException()
+        {
+            // Arrange
+            IEgressExtension extension = CreateExtension();
+
+            // Act & Assert
+            OptionsValidationException exception = await Assert.ThrowsAsync<OptionsValidationException>(
+                () => extension.EgressArtifact(ProviderName, CreateSettings(), WriteFileContent, CancellationToken.None));
```

Comment:
Just double checking that we want to be using `CancellationToken.None` for all of these.

Reply:
In general, I would say it's necessary to pass a real token, but there is no code path that realistically require an async completion.